Helm warning: cannot overwrite table with non table for tolerations (map[])

**Environment:**
 - **Collection**: v2.0.5
 - **Helm**: 3.5.2


When overriding `tolerations` property for Fluentd or OTC, Helm throws warnings about `tolerations` being a map instead of a list.

These warning are due to `tolerations` properties being defined as an empty map `{}` instead of an empty list `[]` in the chart's `values.yaml` file.

Here's an example `values.tolerations.yaml` file:

```yaml
fluentd:
  events:
    statefulset:
      tolerations:
      - key: "example-key"
        operator: "Exists"
        effect: "NoSchedule"

  logs:
    statefulset:
      tolerations:
      - key: "example-key"
        operator: "Exists"
        effect: "NoSchedule"

  metrics:
    statefulset:
      tolerations:
      - key: "example-key"
        operator: "Exists"
        effect: "NoSchedule"

otelagent:
  daemonset:
    tolerations:
    - key: "example-key"
      operator: "Exists"
      effect: "NoSchedule"

otelcol:
  deployment:
    tolerations:
    - key: "example-key"
      operator: "Exists"
      effect: "NoSchedule"

sumologic:
  accessId: dummy
  accessKey: dummy
```

And the result of calling `helm template` with it:

```console
$ helm template collection sumologic/sumologic --values values.tolerations.yaml > /dev/null
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
coalesce.go:200: warning: cannot overwrite table with non table for tolerations (map[])
```

--- a/deploy/helm/sumologic/README.md
+++ b/deploy/helm/sumologic/README.md
@@ -76,7 +76,7 @@ Parameter | Description | Default
 `fluentd.metadata.pluginLogLevel` | Option to give plugin specific log level. | `error`
 `fluentd.logs.enabled` | Flag to control deploying the Fluentd logs statefulsets. | `true`
 `fluentd.logs.statefulset.nodeSelector` | Node selector for Fluentd log statefulset. | `{}`
-`fluentd.logs.statefulset.tolerations` | Tolerations for Fluentd log statefulset. | `{}`
+`fluentd.logs.statefulset.tolerations` | Tolerations for Fluentd log statefulset. | `[]`
 `fluentd.logs.statefulset.affinity` | Affinity for Fluentd log statefulset. | `{}`
 `fluentd.logs.statefulset.podAntiAffinity` | PodAntiAffinity for Fluentd log statefulset. | `soft`
 `fluentd.logs.statefulset.replicaCount` | Replica count for Fluentd log statefulset. | `3`
@@ -145,7 +145,7 @@ Parameter | Description | Default
 `fluentd.logs.default.overrideOutputConf` | Override output section for untagged logs. Leave empty for the default output section. | `Nil`
 `fluentd.metrics.enabled` | Flag to control deploying the Fluentd metrics statefulsets. | `true`
 `fluentd.metrics.statefulset.nodeSelector` | Node selector for Fluentd metrics statefulset. | `{}`
-`fluentd.metrics.statefulset.tolerations` | Tolerations for Fluentd metrics statefulset. | `{}`
+`fluentd.metrics.statefulset.tolerations` | Tolerations for Fluentd metrics statefulset. | `[]`
 `fluentd.metrics.statefulset.affinity` | Affinity for Fluentd metrics statefulset. | `{}`
 `fluentd.metrics.statefulset.podAntiAffinity` | PodAntiAffinity for Fluentd metrics statefulset. | `soft`
 `fluentd.metrics.statefulset.replicaCount` | Replica count for Fluentd metrics statefulset. | `3`
@@ -164,7 +164,7 @@ Parameter | Description | Default
 `fluentd.metrics.overrideOutputConf` | Override output section for metrics. Leave empty for the default output section. | `Nil`
 `fluentd.events.enabled` | If enabled, collect K8s events. | `true`
 `fluentd.events.statefulset.nodeSelector` | Node selector for Fluentd events statefulset. | `{}`
-`fluentd.events.statefulset.tolerations` | Tolerations for Fluentd events statefulset. | `{}`
+`fluentd.events.statefulset.tolerations` | Tolerations for Fluentd events statefulset. | `[]`
 `fluentd.events.statefulset.resources` | Resources for Fluentd log statefulset. | `{"limits":{"cpu":"100m","memory":"256Mi"},"requests":{"cpu":"100m","memory":"256Mi"}}`
 `fluentd.events.statefulset.podLabels` | Additional labels for fluentd events pods. | `{}`
 `fluentd.events.statefulset.podAnnotations` | Additional annotations for fluentd events pods. | `{}`

--- a/deploy/helm/sumologic/values.yaml
+++ b/deploy/helm/sumologic/values.yaml
@@ -327,7 +327,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       ## Acceptable values for podAntiAffinity:
       ## soft: specifies preferences that the scheduler will try to enforce but will not guarantee (Default)
@@ -590,7 +590,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       ## Acceptable values for podAntiAffinity:
       ## soft: specifies preferences that the scheduler will try to enforce but will not guarantee (Default)
@@ -720,7 +720,7 @@ fluentd:
 
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       resources:
         limits:
           memory: 512Mi
@@ -1925,7 +1925,7 @@ otelagent:
   enabled: false
   daemonset:
     nodeSelector: {}
-    tolerations: {}
+    tolerations: []
     resources:
       limits:
         memory: 2Gi
@@ -2027,7 +2027,7 @@ otelagent:
 otelcol:
   deployment:
     nodeSelector: {}
-    tolerations: {}
+    tolerations: []
     replicas: 1
     resources:
       limits:

--- a/tests/upgrade_v2_script/static/fluentd_whole_config.input.yaml
+++ b/tests/upgrade_v2_script/static/fluentd_whole_config.input.yaml
@@ -128,7 +128,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       ## Acceptable values for podAntiAffinity:
       ## soft: specifies preferences that the scheduler will try to enforce but will not guarantee (Default)
@@ -365,7 +365,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       ## Acceptable values for podAntiAffinity:
       ## soft: specifies preferences that the scheduler will try to enforce but will not guarantee (Default)
@@ -490,7 +490,7 @@ fluentd:
 
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       resources:
         limits:
           memory: 512Mi

--- a/tests/upgrade_v2_script/static/fluentd_whole_config.output.yaml
+++ b/tests/upgrade_v2_script/static/fluentd_whole_config.output.yaml
@@ -64,7 +64,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       podAntiAffinity: soft
       replicaCount: 3
@@ -158,7 +158,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       affinity: {}
       podAntiAffinity: soft
       replicaCount: 3
@@ -228,7 +228,7 @@ fluentd:
     enabled: true
     statefulset:
       nodeSelector: {}
-      tolerations: {}
+      tolerations: []
       resources:
         limits:
           memory: 512Mi

--- a/tests/upgrade_v2_script/static/otelcol_image_name.input.yaml
+++ b/tests/upgrade_v2_script/static/otelcol_image_name.input.yaml
@@ -1,7 +1,7 @@
 otelcol:
   deployment:
     nodeSelector: {}
-    tolerations: {}
+    tolerations: []
     replicas: 1
     resources:
       limits:

--- a/tests/upgrade_v2_script/static/otelcol_image_name.output.yaml
+++ b/tests/upgrade_v2_script/static/otelcol_image_name.output.yaml
@@ -1,7 +1,7 @@
 otelcol:
   deployment:
     nodeSelector: {}
-    tolerations: {}
+    tolerations: []
     replicas: 1
     resources:
       limits:

